projects/frontend/src/components/admin-panel/AdminPanel.tsx

```diff
@@ -0,0 +1,178 @@
+`;
+
+const AdminFeatures = styled.div`
+  padding: 12px 30px 20px 30px;
+  box-shadow: 0px 5px 5px 2px ${ props => props.theme.colors.shadowDefault };
+`;
+
+const FeaturesHeader = styled.div`
+  display: flex;
+  align-items: left;
+
+  > h2 {
+    ${ textShadowVisibility }
+    margin: 18px 20px 10px 0px;
+
+    &:first-child {
+      color: ${ props => props.theme.colors.borderDefault };
+    }
+  }
+`;
+
+const CloseBar = styled.div`
+  display: flex;
+  justify-content: center;
+  width: 100%;
+  opacity: 0.5;
+  background-color: ${ props => props.theme.colors.borderRed };
+  border-radius: 8px;
+  padding: 5px 0px;
+  cursor: pointer;
+  box-shadow: 0px 3px 4px ${ props => props.theme.colors.shadowDefault };
+
+  &:hover {
+    opacity: 0.7;
+  }
+
+  > p {
+    color: black;
+    font-weight: bold;
+    opacity: 0.8;
+    margin: 0;
+  }
+`;
+
+const LowerContainer = styled.div`
+  display: flex;
+  overflow-y: hidden;
+  flex-grow: 1;
+
+  @media only screen and (max-width: 780px) {
+    flex-direction: column;
+  }
+`;
+
+const SoundsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  overflow-y: scroll;
+  padding: 20px 30px;
+  border-right: 5px solid ${ props => props.theme.colors.borderDefault };
+
+  > h2 {
+    color: ${ props => props.theme.colors.borderDefault };
+    margin: 5px 0px;
+    
+    ${ textShadowVisibility }
+  }
+
+  &::-webkit-scrollbar {
+    width: 15px;
+    background: ${ props => props.theme.colors.innerB };
+  }
+
+  &::-webkit-scrollbar-thumb {
+    background: ${ props => props.theme.colors.innerA };
+  }
+
+  @media only screen and (max-width: 780px) {
+    border-right: none;
+    border-bottom: 5px solid ${ props => props.theme.colors.borderDefault };
+  }
+`;
+
+interface AdminPanelProps {
+  showAdminPanel: boolean;
+  setShowAdminPanel: (show: boolean) => void;
+  sounds: SWRResponse<Sound[], any, any>;
+  previewRequest: (soundName: string) => Promise<void>
+}
+
+const AdminPanel: FC<AdminPanelProps> = ({ showAdminPanel, setShowAdminPanel, sounds: { data: sounds, error }, previewRequest }) => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const [selectedSound, setSelectedSound] = useState<Sound | null>(null);
+  const [showNotification, setShowNotification] = useState(false);
+  const [notificationProps, setNotificationProps] = useState({ text: '', color: '' });
+
+  const setNotification = useCallback((text: string, color: string) => {
+    setNotificationProps({ text, color });
+    setShowNotification(true);
+    setTimeout(() => setShowNotification(false), 1400);
+  }, []);
+
+  const panelShortcutClose = useCallback((event: KeyboardEvent) => {
+    if (event.key === 'Escape')
+      setShowAdminPanel(false);
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener('keydown', event => panelShortcutClose(event));
+    return document.removeEventListener('keydown', event => panelShortcutClose(event));
+  }, []);
```

Comment:
This is gonna call `removeEventListener()` immediately when the effect runs. It currently works (presumably) because the 2nd argument on line 136 and the 2nd argument on line 137 are not the same function, so it won't find a matching listener to remove.

```suggestion
  useEffect(() => {
    document.addEventListener('keydown', panelShortcutClose, { passive: true });
    return () => document.removeEventListener('keydown', panelShortcutClose);
  }, [panelShortcutClose]);
```